LoopKit/TemporaryPreset.swift

```diff
@@ -0,0 +1,137 @@
+//
+//  TemporaryPreset.swift
+//  Loop
+//
+//  Created by Michael Pangburn on 1/2/19.
+//  Copyright © 2019 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+
+public struct PresetScheduleRepeatOptions: OptionSet, Sendable, Hashable {
+    public let rawValue: UInt8
+
+    public static let none = PresetScheduleRepeatOptions([])
+    public static let sunday = PresetScheduleRepeatOptions(rawValue: 1 << 0)
+    public static let monday = PresetScheduleRepeatOptions(rawValue: 1 << 1)
+    public static let tuesday = PresetScheduleRepeatOptions(rawValue: 1 << 2)
+    public static let wednesday = PresetScheduleRepeatOptions(rawValue: 1 << 3)
+    public static let thursday = PresetScheduleRepeatOptions(rawValue: 1 << 4)
+    public static let friday = PresetScheduleRepeatOptions(rawValue: 1 << 5)
+    public static let saturday = PresetScheduleRepeatOptions(rawValue: 1 << 6)
+
+    public static let allCases: [PresetScheduleRepeatOptions] = [
+        .sunday,
+        .monday,
+        .tuesday,
+        .wednesday,
+        .thursday,
+        .friday,
+        .saturday,
+    ]
```

Comment:
what about weekend and weekdays?

Reply:
Yeah, I think that'd be a nice feature, but not part of the current design.  It's a multiselect, so you can use this to do that.